Cabal/Distribution/InstalledPackageInfo.hs

```diff
@@ -249,6 +252,26 @@ parseExposedModules :: Parse.ReadP r [ExposedModule]
 parseExposedModules = parseOptCommaList parse
 
 -- -----------------------------------------------------------------------------
+-- ABI dependency
+
+data AbiDependency = AbiDependency {
+        depUnitId  :: UnitId,
+        depAbiHash :: AbiHash
+    }
+  deriving (Eq, Generic, Read, Show)
+
+instance Text AbiDependency where
+    disp (AbiDependency uid abi) =
+        disp uid <<>> Disp.char '@' <<>> disp abi
+    parse = do
+        uid <- parse
+        _ <- Parse.char '@'
+        abi <- parse
```

Comment:
I *think* this works with an empty ABI, but would be good to check, since we generate empty ABIs for all compilers other than GHC/GHCJS